clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -193,11 +193,12 @@ public NetworkClientDelegate.PollResult poll(long currentTimeMs) {
         }
         pollTimer.update(currentTimeMs);
         if (pollTimer.isExpired() && !membershipManager.isLeavingGroup()) {
-            logger.warn("Consumer poll timeout has expired. This means the time between " +
+            logger.warn("Consumer poll timeout has expired, exceeded by {} ms. This means the time between " +
```

Comment:
IIUC this is what gets logged when the _heartbeat thread_ notices the consumer has failed to poll in time and dropped out of the group -- so the "time exceeded" is just going to be roughly the max poll interval + the heartbeat interval, no?

I do think it's a great idea to log the amount of time by which the max poll interval was exceeded, but imo the more useful information is how long after the max poll interval the consumer took to actually hit poll again, not how long the heartbeat thread took to notice it.

Reply:
I see what you mean. I think that the background thread will notice more quickly than you said, but this just means the "time exceeded" is going to be very close to max poll interval. The heartbeat request manager checks to see whether it is time to send a heartbeat more regularly than it actually sends a heartbeat.

Maybe enhancing the logging in `HeartbeatRequestManager.resetPollTimer` would be a suitable point. This is where the heartbeat request manager will notice that it has already left the group because of delinquent polling, and rejoins when the next poll occurs. @lianetm that's probably workable I think.

Reply:
Hey, good point, it would actually take this a step further, where indeed should be more useful. As @AndrewJSchofield pointed, the HB manager will notice sooner in practice (even sooner than the HB interval), but we do know when the next poll happens, so can definitely get a more accurate exceed time (in-between calls to poll, which translate to poll events handled in this same manager). On it...thanks for the comments!